cime_config/machines/cmake_macros/crayclanggpu_crusher.cmake

```diff
@@ -0,0 +1,52 @@
+if (compile_threaded)
+  string(APPEND FFLAGS   " -fopenmp")
+  string(APPEND CFLAGS   " -fopenmp")
+  string(APPEND CXXFLAGS " -fopenmp")
+  string(APPEND LDFLAGS  " -fopenmp")
+endif()
+if (DEBUG)
+  string(APPEND CFLAGS   " -O0 -g")
+  string(APPEND FFLAGS   " -O0 -g")
+  string(APPEND CXXFLAGS " -O0 -g")
+  string(APPEND CPPDEFS " -DYAKL_DEBUG")
+endif()
+string(APPEND CPPDEFS " -DFORTRANUNDERSCORE -DNO_R16 -DCPRCRAY")
+string(APPEND FC_AUTO_R8 " -s real64")
+string(APPEND FFLAGS " -f free -N 255 -h byteswapio -em")
+if (NOT compile_threaded)
+  string(APPEND FFLAGS " -M1077")
+endif()
+string(APPEND FFLAGS_NOOPT " -O0")
+set(HAS_F2008_CONTIGUOUS "TRUE")
+string(APPEND LDFLAGS " -Wl,--allow-multiple-definition")
+set(SUPPORTS_CXX "TRUE")
+set(CXX_LINKER "FORTRAN")
+set(MPICC "cc")
+set(MPICXX "hipcc")
+set(MPIFC "ftn")
+set(SCC "cc")
+set(SCXX "hipcc")
+set(SFC "ftn")
+
+if (NOT DEBUG)
+  string(APPEND CFLAGS   " -O2")
+  string(APPEND CXXFLAGS " -O2")
+  string(APPEND FFLAGS   " -O2")
+endif()
+
+if (COMP_NAME STREQUAL elm)
+  string(APPEND FFLAGS " -hfp0")
+endif()
+string(APPEND FFLAGS " -hipa0 -hzero -em -ef -hnoacc")
```

Comment:
Note we don't want to use `-hzero`, as that zeros all stack-allocated arrays. This can lead to uncaught uninitialized-variable bugs.

Reply:
yes, i am consolidating/comparing changes from Az and what scream uses now. hzero is one of many questions.

Reply:
tbc the 1st 2 commits were to bring Az's changes into this branch blindly. the last commit is where i tried to compare everything in his 3 files and in scream's gpu file and put comments.

Reply:
that is, no one is bringing these changes to scream as they are, yet